FirebaseStorage/Sources/FIRStorage.m

```diff
@@ -205,6 +214,50 @@ - (NSString *)description {
   return [NSString stringWithFormat:@"%@ %p: %@", [self class], self, _app];
 }
 
+#pragma mark - Retry time intervals
+
+- (void)setMaxUploadRetryTime:(NSTimeInterval)maxUploadRetryTime {
+  @synchronized(self) {
+    _maxUploadRetryTime = maxUploadRetryTime;
+    _maxUploadRetryInterval =
+        [FIRStorageUtils computeRetryIntervalFromRetryTime:maxUploadRetryTime];
+  }
+}
+
+- (NSTimeInterval)maxDownloadRetryTime {
+  @synchronized(self) {
+    return _maxDownloadRetryTime;
+  }
+}
+
+- (void)setMaxDownloadRetryTime:(NSTimeInterval)maxDownloadRetryTime {
+  @synchronized(self) {
+    _maxDownloadRetryTime = maxDownloadRetryTime;
+    _maxDownloadRetryInterval =
+        [FIRStorageUtils computeRetryIntervalFromRetryTime:maxDownloadRetryTime];
+  }
+}
+
+- (NSTimeInterval)maxUploadRetryTime {
+  @synchronized(self) {
+    return _maxUploadRetryTime;
+  }
+}
+
+- (void)setMaxOperationRetryTime:(NSTimeInterval)maxOperationRetryTime {
+  @synchronized(self) {
+    _maxOperationRetryTime = maxOperationRetryTime;
+    _maxOperationRetryInterval =
+        [FIRStorageUtils computeRetryIntervalFromRetryTime:maxOperationRetryTime];
+  }
+}
+
+- (NSTimeInterval)maxOperationRetryTime {
+  @synchronized(self) {
+    return _maxOperationRetryTime;
+  }
+}
```

Comment:
Do these properties need to be thread-safe? If so, should they all be synchronized with the same lock?

Reply:
I don't think they need to be, but they were before. If we drop this guarantee, this might cause some hard to discover bugs. 

I believe "@synchronized(self)" is the default behavior for synthetic properties. Do you think it makes sense to use different locks per assignment?

Reply:
After thinking about this more, I think it's ok as-is. Using separate locks would decrease contention, but I don't expect these properties to be set or read frequently so it shouldn't be an issue.